Userland/Applications/Browser/History.cpp

```diff
@@ -52,22 +55,28 @@ void History::clear()
     m_current = -1;
 }
 
-const Vector<URL> History::get_back_history()
+void History::update_title(const URL& url, const String& title)
 {
-    Vector<URL> back_history;
+    if (m_items[m_current].url == url)
+        m_items[m_current].title = title;
+}
```

Comment:
This still seems weird. In what case are going to pass in a URL that doesn't match the current history item?!

Reply:
When removing the check I get inconsistent titles when for example going back and forth between two webpages. I'm looking into this because it's really acting in a weird way so there's probably something else going on. The history works fine but the right url isn't stored.

I'm making this a draft for now while I'm looking into it, thanks!